Add documentations about Blowers-Masel reaction (#140)
Cantera/cantera-website: pages/science/reactions.rst

:cti:class:`chebyshev_reaction` entry, or in the YAML format using the
:ref:`Chebyshev <sec-yaml-Chebyshev>` reaction ``type``.

.. _sec-Blowers-Masel:

Blowers-Masel Reactions
-----------------------

> Blowers-Masel rate expression represents the approximation porposed by Blowers

Reviewer: ```suggestion
**New in Cantera 2.6**

The Blowers-Masel rate expression is an approximation proposed by Blowers
```

and Masel [#BlowersMasel2000]_ to automatically scale activation energy

> as we change species' enthalpies. The activation energy estimation can be written

Reviewer: How are species' enthalpies changed? Can you give a one-sentence description of how or why that might happen?

as:

> .. math::
> 
>    E_a = 0 \;\text{if } \Delta H \leq -4 E_a^0
> 
>    E_a = \Delta H \;\text{if } \Delta H \geq 4 E_a^0
> 
>    E_a =  \frac{\left( w + \frac{\Delta H }{2} \right)  (V_P - 2 w + \Delta H) ^2}
>                {V_P^2 - 4 w^2 + \Delta H^2} \;\text{Otherwise}

Reviewer: ```suggestion
.. math::

   E_a = \begin{cases}
      0 & \text{if } \Delta H \leq -4 E_a^0 \\
      \Delta H & \text{if } \Delta H \geq 4 E_a^0 \\
      \frac{\left( w + \frac{\Delta H }{2} \right)  (V_P - 2 w + \Delta H) ^2}
               {V_P^2 - 4 w^2 + \Delta H^2} & \text{Otherwise}
      \end{cases}
```

Reviewer: Please also go into `conf.py` and delete the `filters.typogrify` filter from the list of applied filters. It isn't doing anything useful and just causes problems for us.

where

.. math::

   V_P = 2 w \frac{w + E_a^0}{w - E_a^0},

:math:`w` is the average of the bond dissociation energy of the bond breaking and that being formed,
:math:`E_a^0` is the intrinsic activation energy, and :math:`\Delta H` is the enthalpy change of the reaction.

> Note that the expression is very insensitive to :math:`w` as long as :math:`w \ge 2 E_a^0`, so we can use

Reviewer: ```suggestion
Note that the expression is insensitive to :math:`w` as long as :math:`w \ge 2 E_a^0`, so we can use
```

> an arbitrary high value of :math:`w = 1000 kJ/mol`.

Reviewer: ```suggestion
an arbitrarily high value of :math:`w = 1000\text{ kJ/mol}`.
```

> After :math:`E_a` is evaluated, the reaction rate can be calculated as modified Arrhenius expression

Reviewer: ```suggestion
After :math:`E_a` is evaluated, the reaction rate can be calculated using the modified Arrhenius expression
```

Reviewer: This didn't actually get changed.

.. math::

   k_f = A T^b e^{-E_a / RT}.

Blowers Masel reaction can be defined in the YAML format using the

> :ref:`Blowers-Masel <sec-yaml-Blowers-Masel>` reaction ``type``.

Reviewer: For now, you will probably want to provide the explicit URL link, until 2.6 is released and the automatic reference can be used. The dev documentation URL will be `https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#sec-yaml-blowers-masel`. Please also add a `TODO` in a comment here. Comments can be written in rST as:

```rST
.. TODO: Update the link once version 2.6 is released
```

:cti:class:`chebyshev_reaction` entry, or in the YAML format using the
:ref:`Chebyshev <sec-yaml-Chebyshev>` reaction ``type``.

.. _sec-Blowers-Masel:

Blowers-Masel Reactions
-----------------------

**New in Cantera 2.6**

In some circumstances like thermodynamic sensitivity analysis, or
modeling heterogeneous reactions from one catalyst surface to another,

> the enthalpy of reactions needs to be tweaked. Since reactions' energetics
> is changed, the activation energy of reactions should also be adjusted accordingly
> to provide accurate simulation results. The enthalpy can be modified by changing
> temperature and revising NASA polynomial coefficients in Cantera. To predict the
> activation energy change, the Blowers-Masel rate expression is implemented, which
> is an approximation proposed by Blowers and Masel [#BlowersMasel2000]_ to automatically
> scale activation energy as species' enthalpies are changed in hydrogen transfer reactions.

Reviewer: ```suggestion
the enthalpy change of a reaction (:math:`\Delta H`) can be modified. Due to the change in :math:`\Delta H`,
the activation energy of the reaction must be adjusted accordingly to provide accurate simulation results. To
adjust the activation energy due to changes in the reaction enthalpy, the Blowers-Masel rate expression is
available. This approximation was proposed by Blowers and Masel [#BlowersMasel2000]_ to automatically
scale activation energy as the reaction enthalpy is changed.
```

A couple of notes about this suggestion:
1) It is not clear from the text why the last sentence switches to species enthalpies, when the rest of the paragraph is about reaction enthalpy.
2) Does this only apply to hydrogen transfer reactions? If so, there's no code in the implementation to check and limit that, as far as I know.

Author: 1. Sorry, I mean that the enthalpy of species involved in a reaction change leads to a reaction enthalpy change, I will change it to reaction enthalpy for the consistency.
2. Yes, the paper only justified their approximation using hydrogen transfer reactions, but this approximation could be used for other type of reactions as a first guess if the reaction's enthalpy is changed and we have no idea about the activation energy after the enthalpy is changed.

in the YAML format by specifying the rate constant in the reaction's
:ref:`sticking-coefficient <sec-yaml-interface-reaction>` field.

Surface Blowers-Masel Reactions
-------------------------------

Surface Blowers-Masel Reactions have the same Arrhenius-like rate expression described in
:ref:`Surface Reactions<sec-surface>`, and the activation energy :math:`E_a` is determined
as described in :ref:`Blowers-Masel Reactions<sec-Blowers-Masel>`.

Surface Blowers-Masel reactions can be identified by the presence of surface spcecies and

> the defined :ref:`Blowers-Masel<sec-yaml-surface-Blowers-Masel>` reaction ``type`` in YAML

Reviewer: Likewise here for the URL to the dev docs.

format.

> Note that surface Blowers-Masel reactions also support all the :ref:`additional options <sec-yaml-interface-reaction>`

Reviewer: Likewise here for the URL to the dev docs. Even though this exists in the stable docs, I think it'd be better to link to the dev docs so that the reader has everything they want on one page.

in the YAML format by specifying the rate constant in the reaction's
:ref:`sticking-coefficient <sec-yaml-interface-reaction>` field.

Surface Blowers-Masel Reactions
-------------------------------

> 

Reviewer: ```suggestion

**New in Cantera 2.6**

```

.. TODO: Update the link once version 2.6 is released

Surface Blowers-Masel Reactions have the same Arrhenius-like rate expression described in

> `Surface Reactions <https://cantera.org/science/reactions.html#surface-reactions>`__,

Reviewer: I think this can be a `:ref:` link, using `_sec-surface` that you added here.

and the activation energy :math:`E_a` is determined

> as described in `Blowers-Masel <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#sec-yaml-blowers-masel>`__.

Reviewer: I think this should be a link to the Blowers-Masel documentation on this page, so you can use a `:ref:` link.

Surface Blowers-Masel reactions can be identified by the presence of surface spcecies and

> the defined `Blowers-Masel <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#sec-yaml-surface-blowers-masel>`__
> reaction ``type`` in YAML format.

Reviewer: ```suggestion
the `Blowers-Masel <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#sec-yaml-surface-blowers-masel>`__
reaction ``type`` in a YAML input file.
```

Note that surface Blowers-Masel reactions also support all the `additional options <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#interface>`__

> described in `Surface Reactions <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#interface>`__
> and `sticking-coefficient <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#interface>`__ field in YAML format.

Reviewer: ```suggestion
described in the `Surface Reactions <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#interface>`__
and `sticking-coefficient <https://cantera.org/documentation/dev/sphinx/html/yaml/reactions.html#interface>`__ fields in a YAML input file.
```
